components/script/script_runtime.rs

```diff
@@ -915,11 +915,19 @@ unsafe fn set_gc_zeal_options(_: *mut RawJSContext) {}
 /// enabling an interrupt to be requested
 /// from a thread other than the one running JS using that context.
 #[derive(Clone)]
-pub struct ContextForRequestInterrupt(Arc<Mutex<Option<*mut RawJSContext>>>);
```

Comment:
Is it possible to fix this issue by simply modifying this `Arc` to be an `Rc`?

Reply:
> Is it possible to fix this issue by simply modifying this `Arc` to be an `Rc`?

absolutely yes, I could do that right away. I actually wasn't sure whether this context will only be used in a single-threaded scenario or if it might be accessed from multiple threads in the future. And i ended up choosing the thread safe option. Please let me know if you advice the Rc approach.

Reply:
> Is it possible to fix this issue by simply modifying this `Arc` to be an `Rc`?

hi I just rechecked this, the documentation explicitly states that this needs to be callable from any thread. So I guess it is best to proceed with Send Sync? I've used a custom wrapper with Send Sync here, please let me know if a direct Send Sync is preferred here.